### DISCUSSION TITLE
Getting messages that are expired before a sub is able to receive

### DISCUSSION THREAD

**Discussion by om-parashar** [aws/aws-iot-device-sdk-java-v2#550] [Q&A] [answered]
I have recently implemented Mqtt5Client in our system. 
One thing that i wanted to understand is what happens when a message is expired before a subscriber is there to receive it. As fas as i understood from the documentation that this message will be discarded by the server.
What I was curious in knowing here is that, Is there any way by which my publisher is able to access these expired and discarded messages by the server? If not the publisher, Any method which can just dump these expired messages to some place like SQS?

**Comment by bretambrose** [comment, marked as answer]
I do not believe AWS IoT Core has any support for a dead-letter queue type functionality.  Message expiry is optional/voluntary and if you don't set it I believe you will get the maximum value IoT Core supports.
